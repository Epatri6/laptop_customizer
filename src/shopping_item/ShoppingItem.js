import React from 'react';
import USCurrencyFormat from '../util/USCurrencyFormat';
import './ShoppingItem.css';

export default class ShoppingItem extends React.Component {

  render() {
    const {id, feature, selectedOption} = this.props;
    return (
      <div className="summary__option" key={id}>
      <div className="summary__option__label">{feature} </div>
      <div className="summary__option__value">{selectedOption.name}</div>
      <div className="summary__option__cost">
        {USCurrencyFormat.format(selectedOption.cost)}
      </div>
    </div>
    );
  }

  static defaultProps = {
    id: '',
    feature: '',
    selectedOption: {name: '', cost: 0}
  }

}