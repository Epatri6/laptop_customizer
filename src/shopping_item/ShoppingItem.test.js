import React from 'react';
import ReactDOM from 'react-dom';
import ShoppingItem from './ShoppingItem';
import renderer from 'react-test-renderer';

describe('Testing ShoppingCart Component', () => {

  const testProps = {
    id: '1',
    feature: 'Processor',
    selectedOption: {
      name: '17th Generation Intel Core HB (7 Core with donut spare)',
      cost: 700
    },
  };

  const {id, feature, selectedOption} = testProps;

  //Smoke test
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ShoppingItem id={id} feature={feature} selectedOption={selectedOption}/>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders without props without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ShoppingItem />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  //Snapshot test
  it('renders as expected', () => {
    const tree = renderer
      .create(<ShoppingItem id={id} feature={feature} selectedOption={selectedOption}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
    });
  
  it('renders as expected without props', () => {
    const tree = renderer
      .create(<ShoppingItem />)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
});