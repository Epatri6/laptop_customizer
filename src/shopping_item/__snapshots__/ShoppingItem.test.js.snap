// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ShoppingCart Component renders as expected 1`] = `
<div
  className="summary__option"
>
  <div
    className="summary__option__label"
  >
    Processor
     
  </div>
  <div
    className="summary__option__value"
  >
    17th Generation Intel Core HB (7 Core with donut spare)
  </div>
  <div
    className="summary__option__cost"
  >
    $700.00
  </div>
</div>
`;

exports[`Testing ShoppingCart Component renders as expected without props 1`] = `
<div
  className="summary__option"
>
  <div
    className="summary__option__label"
  >
    
     
  </div>
  <div
    className="summary__option__value"
  >
    
  </div>
  <div
    className="summary__option__cost"
  >
    $0.00
  </div>
</div>
`;
